src/main/world/restaurant/seat.gd

```diff
@@ -6,59 +6,33 @@ contain the shadows itself, since they need to appear behind other objects outsi
 This script contains logic for scaling an injected 'creature shadow' object as the creature grows in size.
 """
 
-"""
-Plays a door chime sound effect, for when a creature enters the restaurant.
-
-Parameter: 'delay' is the delay in seconds before the chime sound plays. The default value of '-1' results in a random
-	delay.
-"""
-func play_door_chime(delay: float = -1) -> void:
-	$Creature.play_door_chime(delay)
-
-
-"""
-Sets the relative position of sound affects related to the restaurant door. Each seat has a different position
-relative to the restaurant's entrance; some are close to the door, some are far away.
-
-Parameter: 'position' is the position of the door relative to this seat, in world coordinates.
-"""
-func set_door_sound_position(position: Vector2) -> void:
-	for chime_sound in $Creature.chime_sounds:
-		chime_sound.position = position
-	for hello_voice in $Creature.hello_voices:
-		hello_voice.position = position
-
-
-"""
-Plays a 'mmm!' creature voice sample, for when a player builds a big combo.
-"""
-func play_combo_voice() -> void:
-	$Creature.play_combo_voice()
-
+var _creature: Creature setget set_creature
+var _door_sound_position: Vector2
 
-"""
-Plays a 'hello!' voice sample, for when a creature enters the restaurant
-"""
-func play_hello_voice() -> void:
-	$Creature.play_hello_voice()
+func set_creature(creature: Creature) -> void:
+	_creature = creature
+	refresh()
 
 
-"""
-Plays a 'check please!' voice sample, for when a creature is ready to leave
-"""
-func play_goodbye_voice() -> void:
-	$Creature.play_goodbye_voice()
+func refresh() -> void:
```

Comment:
Should this be private? Either way, comment it

Reply:
It's referenced outside the class, so it can't be private. I've added a comment.